scipy/sparse/_sputils.py

```diff
@@ -109,13 +109,15 @@ def to_native(A):
 
 
 def getdtype(dtype, a=None, default=None):
-    """Function used to simplify argument processing. If 'dtype' is not
-    specified (is None), returns a.dtype; otherwise returns a np.dtype
-    object created from the specified dtype argument. If 'dtype' and 'a'
-    are both None, construct a data type out of the 'default' parameter.
-    Furthermore, 'dtype' must be in 'allowed' set.
+    """Function used to simplify dtype argument processing.
```

Comment:
The first sentence of a docstring usually describes what the function does. What should it say in this case? I was going to make a suggestion, but I'm not sure!

Reply:
How about:
Determines and checks a dtype based on the arguments provided.

Reply:
I'm trying to document "What would I use this function for", for the benefit of a developer. E.g., "Determine common dtype between input arguments" (not quite) ...

Reply:
How about: 

```
    """Form a supported numpy dtype based on input arguments.

    Returns a valid ``numpy.dtype`` from `dtype` if not None,
    or else ``a.dtype`` if possible, or else the given `default`
    if not None, or else raise a ``TypeError``.

    The resulting ``dtype`` must be in ``supported_dtypes``:
        bool_, int8, uint8, int16, uint16, int32, uint32,
        int64, uint64, longlong, ulonglong, float32, float64,
        longdouble, complex64, complex128, clongdouble
```